q2_taxa/_visualizer.py

```diff
@@ -35,10 +35,20 @@ def barplot(output_dir: str, table: biom.Table, taxonomy: pd.Series,
         raise ValueError('Sample IDs found in the table are missing in the '
                          f'metadata: {ids_not_in_metadata!r}.')
 
+    collapse = True
+    if taxonomy is None:
+        _ids = table.ids('observation')
+        taxonomy = pd.Series(_ids, index=_ids)
+        if not parse_ids:
+            collapse = False
```

Comment:
```suggestion
        if level_delimiter is None:
            collapse = False
        else:
            _ids = table.ids('observation')
            ranks = [';'.join(r.split(level_delimiter)) for r in _ids]
            taxonomy = pd.Series(ranks, index=_ids)
```

Reply:
Just seeing this comment now - sorry. I commented on this below.